### PR TITLE
Allow compile with current Go version

### PR DESCRIPTION
**Pull request by wichert**
With current Go and grpc versions a couple of updates are needed to compile the example. The changes are:

- setup Go modules
- define the Go package in the .proto file
- add `api.UnimplementedHelloServiceServer` to the `HelloServer` struct, which is now required.
- include generate server skeleton code

### CODE REVIEW

**Comment by avinassh** [comment]
wow, I did not know these many changes required for the newer versions! Thank you for the PR. 

I will try running this PR over the weekend and then merge shortly.

File: go/server.go [hunk 1]
```diff
@@ -13,7 +13,9 @@ import (
 	api "github.com/avinassh/grpc-errors/go/hello"
 )
 
-type HelloServer struct{}
+type HelloServer struct {
+	api.UnimplementedHelloServiceServer
+}
```
Comment (avinassh) [inline go/server.go:18]: I had no idea about this! Thank you 🙏

**Review by avinassh — APPROVED** [review]

**Comment by avinassh** [comment]
Thank you so much! 🥳

**Comment by wichert** [comment]
Happy to help! Thank you for the quick review and merge :)

**Pull request merged**